tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -749,6 +750,48 @@ public function test_get_items_status_without_permissions() {
 		}
 	}
 
+	/**
+	 * @ticket 56350
+	 */
+	public function test_get_items_exact_search() {
```

Comment:
For test stability, this test method should instead use a data provider to perform each test.

- The post creation can be moved to `wpSetUpBeforeClass()` and `self::factory()` replaced with `$factory`.
- The test method can accept three parameters: `(string) $search_term`, `(bool) $exact_search`, `(int) $expected`.
- The test method would then look like this:
```php
/**
 * @ticket 56350
 *
 * @dataProvider data_get_items_exact_search
 *
 * @param string $search_term  The search term.
 * @param bool   $exact_search Whether the search is an exact or general search.
 * @param int    $expected     The expected number of matching posts.
 */
public function test_get_items_exact_search( $search_term, $exact_search, $expected ) {
	$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );

	$request->set_param( 'search', $search_term );
	$request->set_param( 'exact_search', $exact_search );

	$response = rest_get_server()->dispatch( $request );

	$this->assertCount( $expected, $response->get_data() );
}
```
- The new data provider, `data_get_items_exact_search()`, can be put below this test method, and can look like this:
```php
/**
 * Data provider for test_get_items_exact_search().
 *
 * @return array[]
 */
public function data_get_items_exact_search() {
	return array(
		'general search, one exact match and one partial match' => array(
			'search_term'  => 'Rye',
			'exact_search' => false,
			'expected'     => 2,
		),
		'exact search, one exact match and one partial match'   => array(
			'search_term'  => 'Rye',
			'exact_search' => true,
			'expected'     => 1,
		),
		'exact search, no match and one partial match'          => array(
			'search_term'  => 'Rye Bread',
			'exact_search' => true,
			'expected'     => 0,
		),
	);
}
```
Additional tests can then be added with the assurance that each test will run, instead of the current situation where a failure in the first test prevents the other tests from running.

Reply:
@costdev The tests changes make sense. @mehulkaklotar ☝️

Reply:
Thanks @costdev @mukeshpanchal27 I have applied your suggestions in the latest commit. Please have a look.

Reply:
@mehulkaklotar Ah thanks for highlighting that issue!

Just one change: The data provider should be placed just below the new test method, not above.

Reply:
@costdev I have placed data provider just below the new test method now. Please have a look. Thanks.